File: Lista fabio (soluções)/ESTRUTURA REPETIÇÃO 04 WHILE/Atividade L/questao05.py
'''
Leia um número, calcule e escreva seu fatorial.
'''

def main():
    numero = int(input("numero: ")) # lendo numero
    fatorial = numero
    c = 1

    if numero == 0: #se o numero for 0, o fatorial será sempre 1
        fatorial = 1
    else: # se não for 0, calcule o fatorial decrementando 1 de "numero" e multiplicando pelo valor anterior de fatorial, que recebe inicialme o proprio numero.
        while c <= numero:
            numero -= 1
            fatorial *= numero
            c+=1
    print(f"fatorial: {fatorial}")


main()

File: Lista fabio (soluções)/ESTRUTURA REPETIÇÃO FOR/Atividade P/questao05.js
// Leia um número, calcule e escreva seu fatorial.

import { get_number } from "./utils/io_utils.js"

function main(){
    let numero = get_number("numero: ")
    let fatorial = numero

    if (numero == 0){
        fatorial = 1 // se o numero for 0, seu fatorial sempre será 1
    }
    else{
        for (let i = 1; i <= numero; i++){
            numero--
            fatorial *= numero
        }
    }
    
    console.log(fatorial)
}

main()